packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -114,6 +115,22 @@ class CupertinoCheckbox extends StatefulWidget {
   /// ```
   final ValueChanged<bool?>? onChanged;
 
+  /// The cursor for a mouse pointer when it enters or is hovering over the
+  /// widget.
+  ///
+  /// If [mouseCursor] is a [WidgetStateMouseCursor],
+  /// [WidgetStateMouseCursor.resolve] is used for the following [WidgetState]s:
+  ///
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///
+  /// When [value] is null and [tristate] is true, [WidgetState.selected] is
+  /// included as a state.
+  ///
+  /// Defaults to [SystemMouseCursors.basic].
+  final MouseCursor? mouseCursor;
```

Comment:
```suggestion
  final WidgetStateProperty<MouseCursor>? mouseCursor;
```

Since we're adding a new property instead of updating, we can add `WidgetStateProperty` mouse cursor.

Reply:
Since Material checkbox is a `MouseCursor` and not a `WidgetStateProperty`, how would .adaptive delegate to CupertinoCheckbox's mouse cursor? Or do we just wrap it with a `WIdgetStateProperty.all` when delegating?

Reply:
I do agree though, especially considering `CheckboxThemeData`'s `mouseCursor` is also a `WidgetStateProperty`.

Reply:
Yes,`WIdgetStatePropertyAll`, for compatibility with adaptive `CheckBox`. 

I would suggest `WidgetStateProperty` for new parameters whenever applicable. Material `CheckBox` and other older widgets contain non-state parameters because state properties doesn't exist back then.

`WidgetStateProperty<MouseCursor>?` new parameter won't need to resolved in the widget and it provide more control over how the mouseCursor can be overridden.